src/helpers/network.rs

```diff
@@ -151,3 +151,122 @@ impl<T: Transport> Network<T> {
         })
     }
 }
+
+#[cfg(all(test, not(feature = "shuttle")))]
+mod tests {
+    use super::*;
+    use crate::{
+        helpers::{CommandEnvelope, HelperIdentity, MESSAGE_PAYLOAD_SIZE_BYTES},
+        test_fixture::transport::InMemoryNetwork,
+    };
+    use futures::stream::StreamExt;
+
+    async fn assert_successful_send(
+        this_network: &Network<impl Transport>,
+        dest_stream: &mut (impl Stream<Item = CommandEnvelope> + Unpin),
+        expected_query_id: QueryId,
+        message_chunks: &MessageChunks,
+        expected_offset: u32,
+    ) {
+        let res = this_network.send(message_chunks.clone()).await;
+        assert!(res.is_ok(), "{res:?}");
+        {
+            let send_offset_tracker = this_network.send_offset_tracker.lock().unwrap();
+            let next_offset = *send_offset_tracker.get(&message_chunks.0).unwrap();
+            assert_eq!(next_offset, expected_offset + 1);
+        }
+        let command = dest_stream.next().await.unwrap();
+        assert_eq!(
+            command.origin,
+            CommandOrigin::Helper(this_network.transport.identity())
+        );
+        if let TransportCommand::StepData {
+            query_id,
+            step,
+            payload,
+            offset,
+        } = command.payload
+        {
+            assert_eq!(query_id, expected_query_id);
+            assert_eq!(step, message_chunks.0.step.clone());
+            assert_eq!(payload, message_chunks.1.clone());
+            assert_eq!(offset, expected_offset);
+        } else {
+            panic!("expected command to be `StepData`, but it was {command:?}")
+        }
+    }
+
+    #[tokio::test]
+    async fn successfully_sends() {
+        let message_chunks = (
+            ChannelId::new(Role::H1, Step::default().narrow("successfully-sends")),
+            vec![0u8; MESSAGE_PAYLOAD_SIZE_BYTES],
+        );
+        let roles = RoleAssignment::new(HelperIdentity::make_three());
+        let this_identity = roles.identity(Role::H2);
```

Comment:
nit: instead of this and that, would be easier to read if you name it `h1_identity`, `h2_identity`

Reply:
done